src/components/ec-inline-input-field/components/read-only/read-only.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <div
+    class="tw-w-full tw-flex tw-justify-between tw-items-center"
+    data-test="ec-inline-input-field-read-only"
+  >
+    <span class="tw-truncate">{{ value }}</span>
+    <button
+      ref="editButton"
+      class="ec-inline-input-field-read-only__action"
+      data-test="ec-inline-input-field-read-only__action"
+      @click="edit"
+      @keydown.enter.space.prevent="edit"
+    >
+      <ec-icon
+        class="ec-inline-input-field-read-only__edit-icon"
+        name="simple-edit"
+        :size="16"
+        type="interactive"
+      />
+    </button>
+  </div>
+</template>
+
+<script>
+import EcIcon from '@/components/ec-icon';
+
+export default {
+  name: 'EcInlineInputFieldReadOnly',
+  components: { EcIcon },
+  props: {
+    value: {
+      default: '',
+      type: String,
+    },
+    gainFocus: {
```

Comment:
The same problem with focus here, this is not a real property. I'd rather use the same approach which I suggested above.
Create a method in this component:
```
focus() {
  this.$nextTick(() => {
     this.$refs.editButton.focus();
   });
}
```
add ref to this component in the parent and call that function from the parent:
```
    cancel(data) {
      this.$emit('cancel');
      this.$refs.readOnly.focus();
    },
```

Reply:
Initially, I used the `gainFocus` data due to the submit process. In the case of cancel process there would be no problem, because it is an action that occurs immediately: the user cancels, the read-only mode is displayed and the button can gain focus immediately. However, when we have a submit, the read-only mode is not shown immediately; we will go to the loading mode and then to the read-only mode, so I thought I needed that data to know, at the moment the component becomes read-only, if the pencil button should gain the focus or not.
Anyway, this functionality has been removed at all because I realized by another of yours comments that it doesn't make sense try to put the focus in the pencil icon programmatically because the submit is an asynchronous operation, and as you said, user can already be working in a different part when we retrieve the response.